datahub-web-react/src/app/shared/tags/TagProfileDrawer.tsx

```diff
@@ -0,0 +1,417 @@
+`;
+
+const ColorPicker = styled.div`
+    position: relative;
+    display: inline-block;
+    margin-top: 1em;
+`;
+
+const ColorPickerButton = styled.div`
+    width: 20px;
+    height: 20px;
+    border: none;
+    border-radius: 50%;
+`;
+
+const ColorPickerPopOver = styled.div`
+    position: absolute;
+    z-index: 100;
+`;
+
+const NoDescriptionText = styled(Typography.Text)`
+    &&& {
+        font-size: 13px;
+        font-weight: 500;
+        font-style: italic;
+        display: block;
+    }
+`;
+
+const DescriptionText = styled(MarkdownViewer)`
+    padding-right: 8px;
+    display: block;
+`;
+
+const EditIcon = styled(EditOutlined)`
+    cursor: pointer;
+`;
+
+const EditDescriptionText = styled(Typography.Text)`
+    &&& {
+        font-size: 13px;
+        font-weight: 500;
+    }
+`;
+
+const HeaderLayout = styled.div`
+    display: flex;
+    justify-content: space-between;
+`;
+
+const StatsBox = styled.div`
+    width: 180px;
+    justify-content: left;
+`;
+
+const StatsLabel = styled(Typography.Text)`
+    &&& {
+        font-size: 13px;
+        font-weight: 600;
+    }
+`;
+
+const EmptyStatsText = styled(Typography.Text)`
+    font-size: 15px;
+    font-style: italic;
+`;
+
+const StatsButton = styled(Button)`
+    padding: 5.6px 0px;
+    margin-top: -4px;
+`;
+
+const ExpandedActions = styled.div`
+    height: 10px;
+`;
+
+const ReadLessText = styled(Typography.Link)`
+    margin-right: 4px;
+`;
+
+const ABBREVIATED_LIMIT = 80;
+
+export const TagProfileDrawer = ({ closeTagProfileDrawer, tagProfileDrawerVisible, urn }: Props) => {
+    const { loading, error, data, refetch } = useGetTagWithColorQuery({ variables: { urn } });
+    const entityRegistry = useEntityRegistry();
+    const history = useHistory();
+    const [updateDescription] = useUpdateDescriptionMutation();
+    const [setTagColorMutation] = useSetTagColorMutation();
+
+    const entityAndSchemaQuery = `tags:"${data?.tag?.properties?.name}" OR fieldTags:"${data?.tag?.properties?.name}" OR editedFieldTags:"${data?.tag?.properties?.name}"`;
+    const entityQuery = `tags:"${data?.tag?.properties?.name}"`;
+    const ownersEmpty = !data?.tag?.ownership?.owners?.length;
+    const description = data?.tag?.properties?.description || '';
+    const hexColor = data?.tag?.properties?.colorHex || '';
+
+    const [showAddModal, setShowAddModal] = useState(false);
+    const [showEditModal, setShowEditModal] = useState(false);
+    const [updatedDescription, setUpdatedDescription] = useState('');
+    const overLimit = removeMarkdown(updatedDescription).length > 80;
+    const [expanded, setExpanded] = useState(!overLimit);
+    const [displayColorPicker, setDisplayColorPicker] = useState(false);
+    const [colorValue, setColorValue] = useState('');
+
+    useEffect(() => {
+        setUpdatedDescription(description);
+        // Manual Color Set-Up, Need to set once it will come from backend
+        setColorValue(hexColor);
+    }, [description, hexColor]);
```

Comment:
these two actions should be in their own useEffects or I think some strange behavior may occur.

Reply:
Fixed and commited.

[https://github.com/linkedin/datahub/pull/4132/commits/f314d27106cd3135e7def6ac9db381a764eb4d73](url)